sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -103,17 +102,19 @@ case class ScalaUDF(
     }
   }
 
-  private def createToScalaConverter(i: Int, dataType: DataType): Any => Any = {
-    if (inputEncoders.isEmpty) {
-      // for untyped Scala UDF
-      CatalystTypeConverters.createToScalaConverter(dataType)
+  private def scalaConverter(i: Int, dataType: DataType): Any => Any = {
```

Comment:
can we keep the name unchanged? If we keep using `createToScalaConverter`, many diff can be avoided.